sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -146,6 +151,15 @@ case class CatalogTable(
   requireSubsetOfSchema(sortColumnNames, "sort")
   requireSubsetOfSchema(bucketColumnNames, "bucket")
 
+  lazy val userSpecifiedSchema: Option[StructType] = if (schema.nonEmpty) {
```

Comment:
I'm not quite sure if it's safe to so, why do we have `CatalogColumn` at the first place?